packages/dd-trace/src/config.js

```diff
@@ -330,6 +330,12 @@ class Config {
       ) : {}
     )
 
+    const DD_TRACE_RAW_PEER_SERVICE_MAPPING = coalesce(process.env.DD_TRACE_PEER_SERVICE_MAPPING,
+      options.peerServiceMapping
+        ? Object.entries(options.peerServiceMapping).map(([key, value]) => `${key}=${value}`).join(',')
```

Comment:
Name isn't consistent.

Reply:
Also, why is it "raw"?

Reply:
I couldn't think of a better name. This configuration is intended to be the "unprocessed" peer service mapping, keeping the value as a string of key/value pair mappings (this is the only way telemetry will accept the configuration). Currently since the JS tracer telemetry flattens the entire config object, the current peer service mapping configuration is being dropped by telemetry intake.

Reply:
Can you clarify what this does? I'm not sure I understand the above. Why would a configuration be needed to get this information?

Reply:
Within the telemetry configuration event, we want to collect the Peer Service Mapping as set by the user (so the actual string mapping of `service_1:remapped_service_1,service_2:remapped_service_2` when a user sets this configuration using the `DD_TRACE_PEER_SERVICE_MAPPING` env variable). However, the JS tracer currently flattens the entire configuration object on this line [here](https://github.com/DataDog/dd-trace-js/blob/04cd89b9dcbe40f73c63b0bc19310715c8446d9e/packages/dd-trace/src/telemetry/index.js#L60C5-L60C36). Since the peer service configuration is parsed into an object [here](https://github.com/DataDog/dd-trace-js/blob/04cd89b9dcbe40f73c63b0bc19310715c8446d9e/packages/dd-trace/src/config.js#L326), our peer service configuration ends up being flattened during telemetry, into something like: 
`configuration_payload = [ ....{"peerServiceMapping.service_1":"remapped_service_1"}, {"peerServiceMapping.service_2":"remapped_service_2"}...] `
Then at telemetry intake these configuration keys end up be dropped. Instead, we should ideally be emitting:
`configuration_payload = [ ....{"peerServiceMapping":"service_1:remapped_service_1,service_2:remapped_service_2"}, ...] `